system/Entity/Entity.php

```diff
@@ -254,11 +254,34 @@ public function hasChanged(?string $key = null): bool
     {
         // If no parameter was given then check all attributes
         if ($key === null) {
-            return $this->original !== $this->attributes;
+            return $this->hasChangedAttributes();
         }
 
         $key = $this->mapProperty($key);
 
+        return $this->hasChangedAttributes($key);
+    }
+
+    /**
+     * Checks a attribute to see if it has changed since the entity
+     * was created. Or, without a parameter, checks if any
+     * attributes have changed.
+     *
+     * @param string|null $key key of $this->attributes
+     */
+    private function hasChangedAttributes(?string $key = null): bool
```

Comment:
Why add new method here ? Why not just add parameter to hasChanged(?string $key = null, $cast = true)

Reply:
boolean parameters are not good practice.

> What is wrong with boolean parameters? | Understand Legacy Code
https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/